idaes/core/util/tables.py

```diff
@@ -224,6 +224,7 @@ def create_stream_table_dataframe(
     """
     stream_attributes = OrderedDict()
     stream_states = stream_states_dict(streams=streams, time_point=time_point)
+    full_keys = []  # List of all rows in dataframe to fill in missing data
```

Comment:
It doesn't seem you are using this in a way that order matters. I would suggest instead making it a set(). This will change lookups ('if k not in full_keys') from linear time to constant time.

Reply:
I am not sure why, but this results in the output being non-deterministic. When  tried doing this, I started seeing the output from the serializer test failing every run because the serialized stream table kept changing.